File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }
        public void InitData() {
            for(int i = 0;i < 5;i++) {
                dataSet11.Tables[0].Rows.Add(new object[] { i, string.Format("FirstName {0}", i), i});
            }
        }
        ColumnHeaderExtender extender;
        private void OnFormLoad(object sender, EventArgs e) {
            InitData();
            extender = new ColumnHeaderExtender(gridView1);
            extender.AddCustomButton();
        }

        private void OnFormClosing(object sender, FormClosingEventArgs e) {
            extender.RemoveCustomButton();
        }

    }
}

File: VB/WindowsApplication3/Main.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors


Namespace DXSample
	Partial Public Class Main
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Sub InitData()
			For i As Integer = 0 To 4
				dataSet11.Tables(0).Rows.Add(New Object() { i, String.Format("FirstName {0}", i), i})
			Next i
		End Sub
		Private extender As ColumnHeaderExtender
		Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			InitData()
			extender = New ColumnHeaderExtender(gridView1)
			extender.AddCustomButton()
		End Sub

		Private Sub OnFormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles MyBase.FormClosing
			extender.RemoveCustomButton()
		End Sub

	End Class
End Namespace
